### DISCUSSION TITLE
Monorepo shared pkgs

### DISCUSSION THREAD

**Discussion by davemccrea** [dokku/dokku#4391] [Q&A] [answered]
Hi! My project uses a monorepo. I came across the dokku-monorepo plugin but doesn't quite work, because my 'frontend' and 'backend' packages reference a 'core' package. Is there a workaround for this? Thanks!

**Comment by josegonzalez** [comment]
What exactly are you trying to do with your app? Have two separate apps with different Procfiles?

**Reply by davemccrea** [reply]
There's 4 packages: 

```
server
store
dashboard
core
```

Server, store and dashboard depend on the core package. The repo is set up with yarn workspaces and works nicely in local development. In production the server runs on Heroku and the store and dashboard are set up with Vercel. But we'd love to transition to dokku! On the dokku side, the server and Svelte frontends are set up as three apps under different subdomains.

My issue is that I don't know how I specify to dokku which package to build when I `git push`. I thought I could use the `dokku-monorepo` plugin but looking at the source code, it appears to copy the package in question to a temporary directory and strips out the other packages. This throws an error during the build stage of course because the server and both web apps depend on the core package (contains types, utils, css, etc). Is it possible to `git push` to dokku and specifiy what package should be built and server?

Hope I'm making sense. Thanks!

Edit: perhaps one solution would be to go use Dockerfile(s) along with `dokku-multi-dockerfile` rather than buildpacks.

**Reply by josegonzalez** [reply]
how do handle deploys of this monorepo to heroku?

**Comment by josegonzalez** [comment, marked as answer]
The proper way to handle this would be to specify alternative paths for specific files (`Procfile`) and set buildpacks on specific apps via the `buildpacks` plugin. This is implemented by https://github.com/dokku/dokku/pull/4502 and will be available in the next minor release (0.25.x).
